src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -55,7 +55,14 @@ public function getTypes(DocType $varType)
 
         $varTypes = [];
         for ($typeIndex = 0; $varType->has($typeIndex); ++$typeIndex) {
-            $varTypes[] = (string) $varType->get($typeIndex);
+            $nestedVarType = $varType->get($typeIndex);
+
+            if ($nestedVarType instanceof Nullable) {
+                $varTypes[] = (string) $nestedVarType->getActualType();
+                $nullable = true;
```

Comment:
Is it expected that this could never be false again the loop? Just asking, I am currently on a phone 😊

Reply:
At least that's the current behaviour. The failing example docblock here was `?string|int` which was expected to reveal `string` and `int` as types both being nullable.

Reply:
the thing is, nullable applies to the whole typehint. Either null is allowed or it is not. But it cannot be both of them.

Reply:
Thanks @stof, that’s what I thought but wanted to be sure 👍🏻